### PR TITLE
MinIO TraceOn in DEBUG mode

### PR DESCRIPTION
**Pull request by mohsha-msft**

### CODE REVIEW

File: cmd/zc_traverser_s3.go [hunk 1]
```diff
@@ -63,6 +69,24 @@ func (t *s3Traverser) IsDirectory(isSource bool) bool {
...
+	// We'll read the buffer, send the data to the log file and reset the buffer for next HTTP Trace.
+	// We can dump the buffer to logfile in one go as well, but we're doing it 1 by 1
+	// 1. Size of buffer stays in control and doesn't grow huge
+	// 2. We want to write trace log before the error log to make sense of what is happening
+	if t.outputStream != nil {
+		traceLog := (*t.outputStream).String()
+		if len(traceLog) > 0 {
+			t.outputStream.Reset()
+			if ste.JobsAdmin != nil {
+				ste.JobsAdmin.LogToJobLog(strings.ReplaceAll(traceLog, "\n\n", "\n"), t.logLevel)
```
Comment (zezha-msft) [inline cmd/zc_traverser_s3.go:84]: anything to scrub?

**Review by zezha-msft — APPROVED** [review]
Please confirm that the necessary scrubbing is done when logging the trace.

File: cmd/zc_traverser_s3.go [hunk 2]
```diff
@@ -214,6 +218,10 @@ func newS3Traverser(credentialType common.CredentialType, rawURL *url.URL, ctx c
 			LogError: glcm.Error,
 		})
 
+	if t.logLevel == pipeline.LogDebug {
```
Comment (nakulkar-msft) [inline cmd/zc_traverser_s3.go:221]: azcopyScanningLogger knows what level to log on, logLevel should need not be inherited from s3Client

File: ste/sourceInfoProvider-S3.go [hunk 3]
```diff
@@ -82,6 +82,9 @@ func newS3SourceInfoProvider(jptm IJobPartTransferMgr) (ISourceInfoProvider, err
 			LogError: func(str string) { p.jptm.Log(pipeline.LogError, str) },
 			Panic:    func(err error) { panic(err) },
 		})
+	if jptm.MinimumLevelToLevel() == pipeline.LogDebug {
```
Comment (nakulkar-msft) [inline ste/sourceInfoProvider-S3.go:85]: Here as well. The appLogger which is part of jptm has enough context to filter logLevel, log level need not be inherited from jptm, you should just set pipeline.LogDebug as level in NewS3HTTPTraceLogger

**Review by nakulkar-msft — CHANGES_REQUESTED** [review]
Inheriting logLevel from traverser or jptm is not necessary as we are always logging at pipeline.LogDebug. Consider defaulting logLevel in NewS3HTTPTraceLogger to pipeline.LogDebug.

**Review by nakulkar-msft — APPROVED** [review]

**Pull request merged**